### PR TITLE
Fix plugin settings

### PR DESCRIPTION
**Pull request by felipebatista**
Hello everybody,

I have experienced some issues trying to setup the plugin on my Redmine **v3.4** instance, similar to https://github.com/zulip/zulip-redmine-plugin/issues/6 & https://github.com/zulip/zulip-redmine-plugin/issues/7

These changes fix the problems with plugin settings

### CODE REVIEW

File: redmine_zulip/lib/zulip_hooks.rb [hunk 1]
```diff
@@ -76,40 +76,51 @@ def zulip_email(project)
...
    def zulip_server()
-       return Setting.plugin_redmine_zulip[:zulip_server]
+     server = Setting.plugin_redmine_zulip["zulip_server"].sub(
+       %r{^https?:(//|\\\\)}i, ""
+     )
+     if server.end_with?("/api")
+       server[0..-5]
+     else
+       server
+     end
```
Comment (timabbott) [inline redmine_zulip/lib/zulip_hooks.rb]: Can you explain this bit of logic?

**Comment by timabbott** [comment]
Thanks for cleaning up this plugin @felipebatista!  It would be great if you can fix the merge conflicts with your now-merged PRs as well as address the comment above (possibly in the form of just adding a code comment explaining where `5` comes from).

**Comment by felipebatista** [comment]
You are very welcome @timabbott! I have just pushed some changes in this bit of code in order to make it more self-explainable, what do you think?

**Comment by timabbott** [comment]
Yes, this is much more readable, thanks @felipebatista!  

Merged.

**Pull request merged**